File: Typescript/src/maxFreqSum.ts
export function maxFreqSum(s: string): number {
	const freq = new Array(26).fill(0);
	let maxVowelFreq = 0;
	let maxConsonantFreq = 0;
	const vowels = new Set(['a', 'e', 'i', 'o', 'u']);
	for (const ch of s) {
		const idx = ch.charCodeAt(0) - 97;
		freq[idx]++;
		if (vowels.has(ch)) {
			maxVowelFreq = Math.max(maxVowelFreq, freq[idx]);
		} else {
			maxConsonantFreq = Math.max(maxConsonantFreq, freq[idx]);
		}
	}
	return maxVowelFreq + maxConsonantFreq;
};

File: C/exercise/maxFreqSum.c

#include "maxFreqSum.h"

int maxFreqSum(char *s)
{
	int freq[26] = {0};
	int max_vowels = 0;
	int max_consonants = 0;
	for (int i = 0; s[i] != '\0'; i++) {
		int index = s[i] - 'a';
		freq[index]++;
		if ((s[i] == 'a') || (s[i] == 'e') || (s[i] == 'i') || (s[i] == 'o') || (s[i] == 'u')) {
			max_vowels = (max_vowels > freq[index]) ? max_vowels : freq[index];
		} else {
			max_consonants = (max_consonants > freq[index]) ? max_consonants : freq[index];
		}
	}
	return max_vowels + max_consonants;
}
